pkg/tfgen/generate_schema.go

```diff
@@ -758,3 +759,92 @@ func (g *Generator) convertExamplesInSchema(spec pschema.PackageSpec) pschema.Pa
 	}
 	return spec
 }
+
+func addExtraHclExamplesToResources(extraExamples []tfbridge.HclExampler, spec *pschema.PackageSpec) error {
+	var err error
+	for _, ex := range extraExamples {
+		token := ex.GetPulumiIdentifier()
+		res, ok := spec.Resources[token]
+		if !ok {
+			err = multierror.Append(err, fmt.Errorf("there is a supplemental HCL example for the resource with token '%s', but no "+
+				"matching resource was found in the schema", token))
+			continue
+		}
+
+		markdown, markdownErr := ex.GetMarkdown()
+		if markdownErr != nil {
+			err = multierror.Append(err, fmt.Errorf("unable to retrieve markdown for example for '%s': %w", token, markdownErr))
+			continue
+		}
+
+		res.Description = appendExample(res.Description, markdown)
+		spec.Resources[token] = res
+	}
+
+	return err
+}
+
+func addExtraHclExamplesToFunctions(extraExamples []tfbridge.HclExampler, spec *pschema.PackageSpec) error {
+	var err error
+	for _, ex := range extraExamples {
+		token := ex.GetPulumiIdentifier()
+		fun, ok := spec.Functions[token]
```

Comment:
can we say `func` here? Also trying and failing to avoid making obnoxious dad jokes about how we at Pulumi are making our own fun

Reply:
No - `func` is a keyword in Go.

Reply:
oh right lol 🤦‍♀️ I still don't want this to say `fun` though!